### ISSUE TITLE
ng-quill with SharePoint AddIns

### ISSUE THREAD

**Issue by ravick4u** [KillerCodeMonkey/ng-quill#140]
I am writing SharePoint 2016 AddIn and trying to use ng-quill with it. When I use it with it I get an error. When I use it with plain HTML page it works but somehow with SharePoint 2016 Addin, it fails.

> SCRIPT5022: [Parchment] Invalid definition
> quill.js (244,9)
> SCRIPT256: This command is not supported.
> quill.js (7451,3)

**Comment by varHarrie** [comment]
https://github.com/quilljs/quill/issues/936

**Comment by KillerCodeMonkey** [comment]
I do not have any experiences with sharepoint addins. Maybe you can explain
it to me and share some code.

Am 15.12.2017 4:09 vorm. schrieb "varHarrie" <notifications@github.com>:

**Comment by ravick4u** [comment]
[Default.txt](https://github.com/KillerCodeMonkey/ng-quill/files/1563165/Default.txt)
SharePoint is Microsoft product.

`

> SharePoint is a web-based, collaborative platform that integrates with Microsoft Office. Launched in 2001,[3] SharePoint is primarily sold as a document management and storage system, but the product is highly configurable and usage varies substantially between organizations.

`

You can see more details here https://en.wikipedia.org/wiki/SharePoint

SharePoint AddIn is client-side development method developed by Microsoft for SharePoint.

Attached is the aspx file(Default.txt) and does not have any serverside code associated with it. This is an aspx file but I could not upload aspx so renamed to txt.

Please let me know if you need any further details.

**Comment by ravick4u** [comment]
@varHarrie Thanks for the response. I believe you are pointing that its related to IE but it's not. I did create plain HTML file and its working in IE.

**Comment by varHarrie** [comment]
I have same issue with IE9, and works fine in IE11. I think it's not compatible with low version of IE.

Another possiblility is related to minification, I learned from [here](https://github.com/KillerCodeMonkey/ng-quill/issues/107#issuecomment-314069324), but not confirmed.

**Comment by ravick4u** [comment]
@varHarrie Thanks for your feedback. When I use ng-quill in pure HTML page it's working in IE and the same is not working with SharePoint AddIn with the same browser. You can take a look at the file which I uploaded in response to @KillerCodeMonkey

**Comment by KillerCodeMonkey** [comment]
@ravick4u any updates here? like i said, i am not an expert with sharepoint. maybe you have to bundle the vendor libraries like quilljs in another way to work with sharepoint?

maybe you should ask in a sharepoint related board or something?

I can not help you here

**Comment by ravick4u** [comment]
@KillerCodeMonkey Thanks for your feedback. I tried couple of senarioes and below is my test results,

Chrome+Pure HTML - Working
IE11+Pure HTML - Working

Chrome+SharePoint - Working
IE11+SharePoint - Working
IE10+SharePoint - Not Working

> SCRIPT5022: [Parchment] Invalid definition
> quill.min.js (7,2544)
> SCRIPT256: This command is not supported.
> quill.min.js (7,148391)

IE9+SharePoint - Not Working

> SCRIPT5007: Unable to get property 'toggle' of undefined or null reference
> quill.min.js (7,147275)

I did the same test with quilljs without angular and got the same results.

So as @varHarrie mentioned the issue is with IE10/IE9 but when I tried with pure HTML without ShaerPoint I don't see that issue.

Because this looks like an issue with quilljs so planning to log this issue here
https://github.com/quilljs/quill
